File: others/Hash Tables/TwoStrings/solve.swift

let s1 = "hello"
let s2 = "world"

func twoStrings(s1: String, s2: String) {
	var s1Ary = Array(s1)
	var s2Ary = Array(s2)
	
	var h: [String: Int] = [:]
	var result = "No"
	
	for i in 0..<max(s1Ary.count, s2Ary.count) {
		if i < s1Ary.count {
			let l1 = String(s1Ary[i])
			if h[l1] == nil {
				h[l1] = 1
			}
			else if let value = h[l1], value < 0 {
				result = "Yes"
				break
			}
		}
		
		if i < s2Ary.count {
			let l2 = String(s2Ary[i])
			if h[l2] == nil {
				h[l2] = -1
			}
			else if let value = h[l2], value > 0 {
				result = "Yes"
				break
			}
		}
	}
	print(result)
}

twoStrings(s1: s1, s2: s2)

File: others/Hash Tables/TwoStrings/solve.py

s1 = "hello"
s2 = "wXrXd"

def twoStrings(s1, s2):
  h = {}
  result = "No"
  for i in range(0, max(len(s1), len(s2))):
    if i < len(s1):
      l1 = s1[i]
      if h.get(l1, None) == None: 
        h[l1] = 1
      elif h[l1] < 0:
        result = "Yes"
        break
    if i < len(s2):
      l2 = s2[i]
      if h.get(l2, None) == None:
        h[l2] = -1
      elif h[l2] > 0:
        result = "Yes"
        break
  print(result)